recipes/vtk/all/conanfile.py

```diff
@@ -0,0 +1,176 @@
+    version = "9.0.1"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://vtk.org/"
+    license = "BSD license"
+    description = "Visualization Toolkit by Kitware"
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    source_subfolder = "source_subfolder"
+    options = {"shared": [True, False], "qt": [True, False], "mpi": [True, False],
+                "fPIC": [True, False], "minimal": [True, False], "ioxml": [True, False],
+                "ioexport": [True, False], "mpi_minimal": [True, False]}
+    default_options = {"shared": False, "qt": False, "mpi": False, "fPIC": False,
+                "minimal": False, "ioxml": False, "ioexport": False, "mpi_minimal": False}
+    topics = ("conan", "VTK", "3D rendering", "2D plotting", "3D interaction", "3D manipulation", 
+                "graphics", "image processing", "scientific visualization", "geometry modeling")
+    short_paths = True
+
+    version_split = version.split('.')
+    short_version = "{}.{}".format(version_split[0], version_split[1])
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("{}-{}".format(self.name, self.version), self.source_subfolder)
+
+    def requirements(self):
+        if self.options.qt:
+            self.requires("qt/5.15.1@bincrafters/stable")
+            self.options["qt"].shared = True
+            if tools.os_info.is_linux:
+                self.options["qt"].qtx11extras = True
+
+    def _system_package_architecture(self):
+        if tools.os_info.with_apt:
+            if self.settings.arch == "x86":
+                return ':i386'
+            elif self.settings.arch == "x86_64":
+                return ':amd64'
+
+        if tools.os_info.with_yum:
+            if self.settings.arch == "x86":
+                return '.i686'
+            elif self.settings.arch == 'x86_64':
+                return '.x86_64'
+        return ""
+
+    def build_requirements(self):
+        pack_names = None
+        if not self.options.minimal and tools.os_info.is_linux:
+            if tools.os_info.with_apt:
+                pack_names = [
+                    "freeglut3-dev",
+                    "mesa-common-dev",
+                    "mesa-utils-extra",
+                    "libgl1-mesa-dev",
+                    "libglapi-mesa",
+                    "libsm-dev",
+                    "libx11-dev",
+                    "libxext-dev",
+                    "libxt-dev",
+                    "libglu1-mesa-dev"]
+
+        if pack_names:
+            installer = tools.SystemPackageTool()
+            for item in pack_names:
+                installer.install(item + self._system_package_architecture())
+
+    def config_options(self):
+        if self.settings.compiler == "Visual Studio":
+            del self.options.fPIC
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
```

Comment:
Please cache the cmake object + you can use `True/False` instead of `"ON"/"OFF"`.

Reply:
Done.
cmake object cached.
I would leave `ON/OFF` vs `True/False` - a lot of other recipes use that notation and IMO it is much more clear that this is actually for cmake not for python

Reply:
Done - Fixed